File: frontend/src/stores/issue.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

interface Issue {
  id: string
  project_id: string
  title: string
  description: string
  status: string
  contact_person: string
  created_at: string
  updated_at: string
}

export const useIssueStore = defineStore('issue', () => {
  const issues = ref<Issue[]>([])
  const currentIssue = ref<Issue | null>(null)

  async function fetchIssues(projectId: string) {
    try {
      // @ts-ignore
      const result = await window.go.main.IssueHandler.ListIssues(projectId)
      issues.value = result || []
    } catch (error) {
      console.error('Failed to fetch issues:', error)
    }
  }

  async function createIssue(projectId: string, title: string, description: string, contactPerson: string) {
    try {
      // @ts-ignore
      const issue = await window.go.main.IssueHandler.CreateIssue(projectId, title, description, contactPerson)
      issues.value.unshift(issue)
      return issue
    } catch (error) {
      console.error('Failed to create issue:', error)
      throw error
    }
  }

  async function updateIssue(id: string, title: string, description: string, status: string, contactPerson: string) {
    try {
      // @ts-ignore
      await window.go.main.IssueHandler.UpdateIssue(id, title, description, status, contactPerson)
      const index = issues.value.findIndex(i => i.id === id)
      if (index !== -1) {
        issues.value[index] = { ...issues.value[index], title, description, status, contact_person: contactPerson }
      }
    } catch (error) {
      console.error('Failed to update issue:', error)
      throw error
    }
  }

  async function deleteIssue(id: string) {
    try {
      // @ts-ignore
      await window.go.main.IssueHandler.DeleteIssue(id)
      issues.value = issues.value.filter(i => i.id !== id)
      if (currentIssue.value?.id === id) {
        currentIssue.value = null
      }
    } catch (error) {
      console.error('Failed to delete issue:', error)
      throw error
    }
  }

  async function updateStatus(id: string, status: string) {
    try {
      // @ts-ignore
      await window.go.main.IssueHandler.UpdateStatus(id, status)
      const index = issues.value.findIndex(i => i.id === id)
      if (index !== -1) {
        issues.value[index] = { ...issues.value[index], status }
      }
    } catch (error) {
      console.error('Failed to update status:', error)
      throw error
    }
  }

  function setCurrentIssue(issue: Issue) {
    currentIssue.value = issue
  }

  return {
    issues,
    currentIssue,
    fetchIssues,
    createIssue,
    updateIssue,
    deleteIssue,
    updateStatus,
    setCurrentIssue
  }
})

File: internal/handlers/issue.go
package handlers

import (
	"fmt"
	"time"

	"Pebble/internal/database"

	"github.com/google/uuid"
)

type IssueHandler struct{}

func NewIssueHandler() *IssueHandler {
	return &IssueHandler{}
}

func (h *IssueHandler) CreateIssue(projectID, title, description, contactPerson string) (*database.Issue, error) {
	issueID := uuid.New().String()

	_, err := database.DB.Exec(
		`INSERT INTO issues (id, project_id, title, description, contact_person) 
		 VALUES (?, ?, ?, ?, ?)`,
		issueID, projectID, title, description, contactPerson,
	)
	if err != nil {
		return nil, fmt.Errorf("failed to create issue: %w", err)
	}

	return h.GetIssue(issueID)
}

func (h *IssueHandler) GetIssue(id string) (*database.Issue, error) {
	issue := &database.Issue{}
	err := database.DB.QueryRow(
		`SELECT id, project_id, title, description, status, contact_person, created_at, updated_at 
		 FROM issues WHERE id = ?`, id,
	).Scan(&issue.ID, &issue.ProjectID, &issue.Title, &issue.Description,
		&issue.Status, &issue.ContactPerson, &issue.CreatedAt, &issue.UpdatedAt)
	if err != nil {
		return nil, fmt.Errorf("failed to get issue: %w", err)
	}
	return issue, nil
}

func (h *IssueHandler) ListIssues(projectID string) ([]database.Issue, error) {
	rows, err := database.DB.Query(
		`SELECT id, project_id, title, description, status, contact_person, created_at, updated_at 
		 FROM issues WHERE project_id = ? ORDER BY created_at DESC`, projectID,
	)
	if err != nil {
		return nil, fmt.Errorf("failed to list issues: %w", err)
	}
	defer rows.Close()

	var issues []database.Issue
	for rows.Next() {
		var i database.Issue
		err := rows.Scan(&i.ID, &i.ProjectID, &i.Title, &i.Description,
			&i.Status, &i.ContactPerson, &i.CreatedAt, &i.UpdatedAt)
		if err != nil {
			return nil, fmt.Errorf("failed to scan issue: %w", err)
		}
		issues = append(issues, i)
	}
	return issues, nil
}

func (h *IssueHandler) UpdateIssue(id, title, description, status, contactPerson string) error {
	_, err := database.DB.Exec(
		`UPDATE issues SET title = ?, description = ?, status = ?, contact_person = ?, updated_at = ? WHERE id = ?`,
		title, description, status, contactPerson, time.Now(), id,
	)
	if err != nil {
		return fmt.Errorf("failed to update issue: %w", err)
	}
	return nil
}

func (h *IssueHandler) DeleteIssue(id string) error {
	_, err := database.DB.Exec(`DELETE FROM issues WHERE id = ?`, id)
	if err != nil {
		return fmt.Errorf("failed to delete issue: %w", err)
	}
	return nil
}

func (h *IssueHandler) UpdateStatus(id, status string) error {
	_, err := database.DB.Exec(
		`UPDATE issues SET status = ?, updated_at = ? WHERE id = ?`,
		status, time.Now(), id,
	)
	if err != nil {
		return fmt.Errorf("failed to update status: %w", err)
	}
	return nil
}
